lang/rust/avro/src/de.rs

```diff
@@ -676,6 +676,49 @@ mod tests {
 
     use super::*;
 
+    #[derive(PartialEq, Eq, Serialize, Deserialize, Debug)]
+    pub struct StringEnum {
+        pub source: String,
+    }
+
+    #[test]
+    fn avro_xxxx_decode_enum() {
+        let schema_content = r#"
+{
+  "name": "AccessLog",
+  "namespace": "com.clevercloud.accesslogs.common.avro",
+  "type": "record",
+  "fields": [
+    {
+      "name": "source",
+      "type": {
+        "type": "enum",
+        "name": "SourceType",
+        "items": "string",
+        "symbols": ["SOZU", "HAPROXY", "HAPROXY_TCP"]
+      }
+    }
+  ]
+}
+"#;
+
+        let schema = crate::Schema::parse_str(schema_content).unwrap();
+        let data = StringEnum { source: "SOZU".to_string() };
```

Comment:
Let's add one more test case with a value that is not one of the enum symbols too!

Reply:
It is done :heavy_check_mark: